### ISSUE TITLE
CentOS < 7 Security Advisory Issues

### ISSUE THREAD

**Issue by wied03** [wied03/centos-package-cron#5]
From Andrew V:

I'm running CentOS 6.6 with bash-4.1.2-29.el6.x86_64. I get this in the email as the very first notification:

```
Advisory ID: CESA-2014--1293
Severity: Critical
Packages:
* bash-4.1.2-29.el6
References:
* https://rhn.redhat.com/errata/RHSA-2014-1293.html
* http://lists.centos.org/pipermail/centos-announce/2014-September/020582.html
* http://lists.centos.org/pipermail/centos-announce/2014-September/020583.html
* http://lists.centos.org/pipermail/centos-announce/2014-September/020585.html
* http://lists.centos.org/pipermail/centos-announce/2014-September/020650.html
```

However there is nothing wrong with this version of bash. The RHEL link there provides an older version of bash as the solution, and notes that it's outdated by https://rhn.redhat.com/errata/RHBA-2014-1503.html. This latest errata page notes that I am running the latest version.

**Comment by wied03** [comment]
When Steve's advisory list is checked, the following process happens:
1. Check if any package name in the advisory list is installed in the system
2. Check if the currently installed version is older than **any** of the versions mentioned in the advisory
3. Check if the OS release matches the list of releases in the advisory

The problem here appears to be with 2 above. Even though 4.1.2-15 (and above) is considered a fix for CentOS6, the 4.2.45-5 in the list, which is really for CentOS7, is causing the advisory to match. I probably haven't noticed in since I run CentOS7. The XML blurb from Steve looks like this:
```
<CESA-2014--1293 description="Not available" from="centos-announce@centos.org" issue_date="2014-09-24 15:07:20" multirelease="1" notes="Not available" product="CentOS Linux" references="https://rhn.redhat.com/errata/RHSA-2014-1293.html http://lists.centos.org/pipermail/centos-announce/2014-September/020582.html http://lists.centos.org/pipermail/centos-announce/2014-September/020583.html http://lists.centos.org/pipermail/centos-announce/2014-September/020585.html http://lists.centos.org/pipermail/centos-announce/2014-September/020650.html" release="4" severity="Critical" solution="Not available" synopsis="Critical CentOS bash Security Update" topic="Not available" type="Security Advisory">
    <os_arch>i386</os_arch>
    <os_arch>i686</os_arch>
    <os_arch>x86_64</os_arch>
    <os_release>5</os_release>
    <os_release>6</os_release>
    <os_release>7</os_release>
    <packages>bash-3.2-33.el5.1.i386.rpm</packages>
    <packages>bash-3.2-33.el5.1.src.rpm</packages>
    <packages>bash-3.2-33.el5.1.x86_64.rpm</packages>
    <packages>bash-4.1.2-15.el6_5.1.i686.rpm</packages>
    <packages>bash-4.1.2-15.el6_5.1.src.rpm</packages>
    <packages>bash-4.1.2-15.el6_5.1.x86_64.rpm</packages>
    <packages>bash-4.2.45-5.el7_0.2.src.rpm</packages>
    <packages>bash-4.2.45-5.el7_0.2.x86_64.rpm</packages>
    <packages>bash-doc-4.1.2-15.el6_5.1.i686.rpm</packages>
    <packages>bash-doc-4.1.2-15.el6_5.1.x86_64.rpm</packages>
    <packages>bash-doc-4.2.45-5.el7_0.2.x86_64.rpm</packages>
  </CESA-2014--1293>
```

Here are the fix options as I see them:
1. Try and get Steve to add release/package hierarchy to the XML like the RHN page shows.
2. Match on subversion/release (as in green light 4.1.2-29 because the version matches but the release, 29, is greater than or equal to 15).
3. Have the parser look at [centos4,el5, el6_5, el7] in the filename of the package to decide which packages are relevant

Pros/cons of each:
1. Given Steve is just parsing the CentOS announce email list and that does not have hierarchy, this is probably not an option.
2. Not too invasive, but can we always count on a major security fix just being a minor release update?
3. Probably more robust than 2, but a little more of a pain to code. I don't have the cycles to do regex experimenting right this instant, but can we always count on that info being in the filename?

**Comment by wied03** [comment]
Regarding option 3 above, https://fedoraproject.org/wiki/Packaging:DistTag implies that the dist tag is mandatory and it can be counted on

**Comment by andrewvoce** [comment]
I think option three would be best. If I ran CentOS 7 this wouldn't be an issue. I know a few people that are hesitant to move to CentOS 7 due to systemd and whatever philosophical battles they're raging, but I'm not sure how widespread that is. If I am one of the very few that is still running el6 or has trouble with this issue, then maybe this isn't worth looking into and upgrading to el7 is the better answer.

**Comment by wied03** [comment]
@andrewvoce, try checking out the https://github.com/wied03/centos-package-cron/tree/bugfixes/5_oldercentos branch and see if it works properly for you. Remove /var/lib/centos-package-cron/already_annoyed.sqlite before you run it to get a good test.

**Comment by andrewvoce** [comment]
That worked great! Would it be easy to add an opt to direct output to std out instead of or in addition to email?

**Comment by wied03** [comment]
Done. released as 1.0.5.1.
